azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/presentation/ui/reduxviewmodel/ReduxViewModel.kt

```diff
@@ -0,0 +1,58 @@
+package com.azure.android.communication.ui.chat.presentation.ui.reduxviewmodel
+
+import com.azure.android.communication.ui.chat.redux.AppStore
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+/**
+ * Redux View Model
+ *
+ * Listens to the Store and Emits ViewModels
+ *
+ *  T = Type of the Store store
+ *  M = Type of the ViewModel
+
+ * Params:
+ *   builder: (T) -> M
+ *   onChanged: (M) -> Unit
+ *   coroutineScope: The lifecycle scope to run this under
+ *   store: The Store you want to listen to
+ */
+internal class ReduxViewModel<T, M : Any>(
```

Comment:
Is there a reason for this viewmodel to have separate viewmodel directory?
Could this viewmodel go under the viewmodel directory?

Reply:
This isn't a ViewModel, it's the VIewModel Generation Tool.

I.e. It Makes View Models by listening to Redux. I can rename to ReduxViewModelGenerator

Reply:
Renamed for clarity